### DISCUSSION TITLE
Cannot install homebrew on Ubuntu 20.04

### DISCUSSION THREAD

**Discussion by rivernet1** [Homebrew/discussions#3275] [Getting started] [answered]
```
==> Tapping homebrew/core
remote: Enumerating objects: 1193528, done.
remote: Counting objects: 100% (245/245), done.
remote: Compressing objects: 100% (108/108), done.
remote: Total 1193528 (delta 152), reused 227 (delta 137), pack-reused 1193283
Receiving objects: 100% (1193528/1193528), 477.35 MiB | 10.97 MiB/s, done.
Resolving deltas: 100% (826816/826816), done.
From https://github.com/Homebrew/homebrew-core
 * [new branch]              master     -> origin/master
HEAD is now at 8741ca4ac62 commitizen: update 2.27.0 bottle.
==> Downloading https://ghcr.io/v2/homebrew/portable-ruby/portable-ruby/blobs/sha256:cd7fffb18ef9338baa670fc5e8fce99b0e3cc0f0fd7627bcbb56f3c8d54161d4
Warning: Failed to create the file
Warning: /home/anthony/.cache/Homebrew/portable-ruby-2.6.8.x86_64_linux.bottle.
Warning: tar.gz.incomplete: Permission denied
curl: (23) Failure writing output to destination

==> Downloading https://github.com/Homebrew/homebrew-portable-ruby/releases/download/2.6.8/portable-ruby-2.6.8.x86_64_linux.bottle.tar.gz
Warning: Failed to create the file
Warning: /home/anthony/.cache/Homebrew/portable-ruby-2.6.8.x86_64_linux.bottle.
Warning: tar.gz.incomplete: Permission denied
curl: (23) Failure writing output to destination

Error: Failed to download ruby from the following locations:
  - https://ghcr.io/v2/homebrew/portable-ruby/portable-ruby/blobs/sha256:cd7fffb18ef9338baa670fc5e8fce99b0e3cc0f0fd7627bcbb56f3c8d54161d4
  - https://github.com/Homebrew/homebrew-portable-ruby/releases/download/2.6.8/portable-ruby-2.6.8.x86_64_linux.bottle.tar.gz

Do not file an issue on GitHub about this; you will need to figure out for
yourself what issue with your internet connection restricts your access to
GitHub (used for Homebrew updates and binary packages).
Error: Failed to install Homebrew Portable Ruby and cannot find another Ruby 2.6.8!
If there's no Homebrew Portable Ruby available for your processor:
- install Ruby 2.6.8 with your system package manager (or rbenv/ruby-build)
- make it first in your PATH
- try again

Failed during: /home/linuxbrew/.linuxbrew/bin/brew update --force --quiet
```

**Comment by SMillerDev** [comment]
```
Warning: Failed to create the file
Warning: /home/anthony/.cache/Homebrew/portable-ruby-2.6.8.x86_64_linux.bottle.
Warning: tar.gz.incomplete: Permission denied
curl: (23) Failure writing output to destination
```

You should check the permissions of that directory, it seems the installer can't write to the cache

**Comment by rivernet1** [comment]
I’ll look into that.  Thanks for the suggestion.


From: Sean Molenaar ***@***.***>
Reply-To: Homebrew/discussions ***@***.***>
Date: Monday, May 16, 2022 at 11:48 AM
To: Homebrew/discussions ***@***.***>
Cc: rivernet1 ***@***.***>, Author ***@***.***>
Subject: Re: [Homebrew/discussions] Cannot install homebrew on Ubuntu 20.04 (Discussion #3275)


Warning: Failed to create the file

Warning: /home/anthony/.cache/Homebrew/portable-ruby-2.6.8.x86_64_linux.bottle.

Warning: tar.gz.incomplete: Permission denied

curl: (23) Failure writing output to destination

You should check the permissions of that directory, it seems the installer can't write to the cache

**Comment by rivernet1** [comment]
Thanks for the reply. I’ve tried what you suggested and had the same result.

I might have to do a diskcheck as this installation has been running “iffy”, here as of late.

Anthony


From: Sean Molenaar ***@***.***>
Reply-To: Homebrew/discussions ***@***.***>
Date: Monday, May 16, 2022 at 11:48 AM
To: Homebrew/discussions ***@***.***>
Cc: rivernet1 ***@***.***>, Author ***@***.***>
Subject: Re: [Homebrew/discussions] Cannot install homebrew on Ubuntu 20.04 (Discussion #3275)


Warning: Failed to create the file

Warning: /home/anthony/.cache/Homebrew/portable-ruby-2.6.8.x86_64_linux.bottle.

Warning: tar.gz.incomplete: Permission denied

curl: (23) Failure writing output to destination

You should check the permissions of that directory, it seems the installer can't write to the cache

**Reply by gromgit** [reply]
What's the output of `ls -ld /home/anthony/.cache/Homebrew`?

**Comment by rivernet1** [comment]
anthony  ubuntu-orion-1  ~  ls -ld /home/anthony/.cache/Homebrew
drwxrwxr-x 2 anthony anthony 4.0K 2022-05-16 11:19 /home/anthony/.cache/Homebrew


From: Adrian Ho ***@***.***>
Reply-To: Homebrew/discussions ***@***.***>
Date: Thursday, May 19, 2022 at 8:22 AM
To: Homebrew/discussions ***@***.***>
Cc: rivernet1 ***@***.***>, Author ***@***.***>
Subject: Re: [Homebrew/discussions] Cannot install homebrew on Ubuntu 20.04 (Discussion #3275)


What's the output of ls -ld /home/anthony/.cache/Homebrew?

**Reply by gromgit** [reply]
And what's the output of `ls -ld /home/anthony/.cache/Homebrew/portable-ruby-2.6.8.x86_64_linux.bottle.tar.gz.incomplete`?

**Comment by rivernet1** [comment, marked as answer]
Thanks for the help. I just nuked the installation and started fresh. Everything works like a champ now.

Get Outlook for iOS<https://aka.ms/o0ukef>. Please allow ample time for a proper reply, as this is from my mobile device.
________________________________
